src/Storages/MergeTree/MergeTreePartition.cpp

```diff
@@ -261,8 +261,11 @@ String MergeTreePartition::getID(const Block & partition_key_sample) const
     hash.get128(hash_data);
     result.resize(32);
     for (size_t i = 0; i < 16; ++i)
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
```

Comment:
`if constexpr (std::endian::native == std::endian::big)` would be nicer

Reply:
`if constexpr (std::endian::native == std::endian::big)` only works when it is in a template function. Since getID() is not a template function, using this way would cause compiling error.